CoffeeChat/Controller/DemographicsViewController.swift

```diff
@@ -165,4 +160,70 @@ class DemographicsViewController: UIViewController {
             make.bottom.equalTo(view.safeAreaLayoutGuide).inset(nextBottomPadding)
         }
     }
+
+    private func updateFieldConstraints(fieldView: UIView, height: CGFloat) {
+        fieldView.snp.updateConstraints { make in
+            make.height.equalTo(height)
+        }
+    }
+}
+
+extension DemographicsViewController: OnboardingSearchViewDelegate {
+
+    private func numberFieldsSelected() -> Int {
+        var count = 0
+        for field in fieldsEntered where field { count+=1 }
+        return count
+    }
+
+    internal func updateSelectedFields(tag: Int, isSelected: Bool) {
+        fieldsEntered[tag] = isSelected
+        nextButton.isEnabled = numberFieldsSelected() == 4
+        nextButton.backgroundColor = nextButton.isEnabled ? .backgroundOrange : .inactiveGreen
+        nextButton.layer.shadowColor = nextButton.isEnabled ? UIColor.black.cgColor : .none
+        nextButton.layer.shadowOffset = nextButton.isEnabled ? CGSize(width: 0.0, height: 2.0) : CGSize(width: 0.0, height: 0.0)
+        nextButton.layer.shadowOpacity = nextButton.isEnabled ? 0.15 : 0
+        nextButton.layer.shadowRadius = nextButton.isEnabled ? 2 : 0
```

Comment:
ternaries are usually preferred but in this case where its multiple in a row i think its more readable to just separate w/ an if block:
```
if nextButton.isEnabled {

} else {

}
```
it also just makes it easier to add more changes in the future that depend on `nextButton.isEnabled`